llama_index/evaluation/batch_runner.py

```diff
@@ -0,0 +1,78 @@
+import asyncio
+from typing import Any, Dict, List, Optional, Tuple
+
+from llama_index.evaluation.base import BaseEvaluator, EvaluationResult
+from llama_index.indices.query.base import BaseQueryEngine
+from llama_index.response.schema import Response
+
+
+class BatchEvalRunner:
+    def __init__(
+        self,
+        evaluators: Dict[str, BaseEvaluator],
+        workers: int = 2,
+    ):
+        self.evaluators = evaluators
+        self.workers = workers
+
+    async def aevaluate_queries(
+        self,
+        query_engine: BaseQueryEngine,
+        queries: List[str] = None,
+        query_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> Dict[str, List[EvaluationResult]]:
+        query_kwargs = query_kwargs or {}
+
+        semaphore = asyncio.Semaphore(self.workers)
```

Comment:
fancy!

Reply:
curious why is this necessary?

Reply:
Just a form of handling rate limiting (without actually handling it)

Prevents too many requests from going through at once